### PR TITLE
#142 Own classloader

### LINKED ISSUE #142: Feature Request:  Use separate class loaders
Per #79, the xlconnect package conflicts with the xlsx package because they share a Java dependency.  However, with s-u/rJava#250, it should now be possible to prevent this.  Would it be possible to apply this enhancement from rJava?

### PR DESCRIPTION
**Pull request by spoltier**
Will eventually close #142, opening to get checks

### CODE REVIEW

**Comment by spoltier** [comment]
@RolandASc @martinstuder @riccardoporreca  - this implies raising the minimum version of R to 3.6.0. As such we may want to delay merging this, not sure what you think.

File: R/J.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+#############################################################################
+#
+# Override rJava's J() to use the dedicated classloader, as advised by
+# Simon Urbanek in the rJava CHANGELOG (https://cran.r-project.org/package=rJava/NEWS)
+# 
+# Author: Simon Poltier, Mirai Solutions GmbH
+#
+#############################################################################
+J <- function(...) rJava::J(..., class.loader=.rJava.class.loader)
```
Comment (spoltier) [inline R/J.R:29]: Not sure if I should include this in a different way, since it is copied straight from the NEWS file.

**Comment by martinstuder** [comment]
From my point of view it's ok to raise the minimum R version requirement to 3.6 given that 4.x is already out

**Comment by spoltier** [comment]
@RolandASc resolved conflicts

**Review by RolandASc — COMMENTED** [review]
Thanks @spoltier .
I have just two general comments:

- should `.devcontainer` be added to `.Rbuildignore`?
- there's a note on rJava 1.0-5 https://cran.r-project.org/web/packages/rJava/NEWS. is that going to be a problem for us?

**Comment by spoltier** [comment]
> Thanks @spoltier . I have just two general comments:
> 
> * should `.devcontainer` be added to `.Rbuildignore`?
> * there's a note on rJava 1.0-5 https://cran.r-project.org/web/packages/rJava/NEWS. is that going to be a problem for us?

.devcontainer and the corresponding Dockerfile are meant for use with visual studio code, I think it would be useful to have - It would then be possible to make a small change and run some tests without having R and all the necessary packages set up, only docker and vs code being required. 
I can add something to the README if you want. VS code will detect this folder and ask if the user wants to open the workspace in a container.
I needed it because I had issues switching java versions on my Mirai laptop.

In terms of the note on rJava, we would have noticed the issue in our checks on master already if it was going to be a problem.

**Pull request merged**